x-pack/legacy/plugins/actions/server/builtin_action_types/slack.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Joi from 'joi';
+import { IncomingWebhook } from '@slack/webhook';
+
+import { ActionType, ActionTypeExecutorOptions } from '../types';
+
+// TODO: figure out the right way type-safe way to mock slack's IncomingWebhook class
+let IncomingWebhookImpl: any = IncomingWebhook;
+
+// for testing
+export function setIncomingWebhookImpl(incomingWebHook: any = IncomingWebhook): void {
+  IncomingWebhookImpl = incomingWebHook;
+}
+
+const CONFIG_SCHEMA = Joi.object().keys({
+  webhookUrl: Joi.string().required(),
+});
+
+const PARAMS_SCHEMA = Joi.object().keys({
+  message: Joi.string().required(),
+});
+
+export const actionType: ActionType = {
+  id: 'kibana.slack',
```

Comment:
We may want to follow the convention we agreed on here? (`.slack` or something in this case?) https://github.com/elastic/kibana/pull/39093#discussion_r294928213

Reply:
whatcha thinking then @mikecote ?  `.kibana.x-pack.slack` seems pretty good; anyone using this programmatically would then know it's an x-pack feature, for instance.

Reply:
I have no strong preference in regards to this, the way I have the docs written up is anything that begins with `.` is a built in action type. That leaves anything after the `.` up to us :).

I don't think it's necessary to include x-pack since the plugin is an x-pack feature, we can do `.slack` or `.kibana.slack` or if you have something better I'm good with that too.

We can always change the docs if ever this convention seems too weird.

Reply:
Probably no great reason to have a `kibana` prefix then either, will go with `.slack`.